include/s3select_functions.h

```diff
@@ -1313,6 +1320,106 @@ struct _fn_string : public base_function
   }
 };
 
+struct _fn_trim : public base_function {
+
+    std::string input_string;
+    value v_remove;
+    value v_input;
+
+    void removespacebothside(std::string& s,const char* remove_chars)
+    {
+      boost::trim_right_if(s,boost::is_any_of(remove_chars));
+      boost::trim_left_if(s,boost::is_any_of(remove_chars));
+    }
+
+    bool operator()(bs_stmt_vec_t* args, variable* result)
+    {
+    	bs_stmt_vec_t::iterator iter = args->begin();
+    	int args_size = args->size();
+    	base_statement* str = *iter;
+        v_input = str->eval();
+        if(v_input.type != value::value_En_t::STRING) {
+            throw base_s3select_exception("content is not string");
+        }
+        v_remove = " "; 
```

Comment:
v_remove could set in constructor, in case of using default remove-char, it set only once.